hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java

```diff
@@ -293,7 +294,7 @@ boolean initInvocationTarget() {
         }
 
         targetMember = nodeEngine.getClusterService().getMember(invTarget);
-        if (!isJoinOperation(op) && targetMember == null) {
+        if (!(isJoinOperation(op) || isWanReplicationOperation(op)) && targetMember == null) {
```

Comment:
it would be great to include this explanation in a commit message. just for a future reference. 
currently the commit message doesn't say anything about it.